codex-process-data-transfer/src/main/java/de/netzwerk_universitaetsmedizin/codex/processes/data_transfer/client/fhir/AbstractFhirClient.java

```diff
@@ -653,6 +653,7 @@ public void updatePatient(Patient patient)
 		Objects.requireNonNull(patient, "patient");
 
 		String id = patient.getIdElement().toVersionless().getValue();
+		patient.setId(id);
```

Comment:
But it isn't. If the version is set, the `If-Match` header is set by HAPI client (in our and the current version https://github.com/hapifhir/hapi-fhir/blob/master/hapi-fhir-client/src/main/java/ca/uhn/fhir/rest/client/method/MethodUtil.java#L233). This will result in a incorrectly set `If-Match`-Header if the server sent a `Etag`-Header in the GET-request, that is not `"version"`. E.g. Blaze will set the `Etag`-Header to `W/"version"` and will refuse to update the patient.

Reply:
See here https://www.hl7.org/fhir/http.html#concurrency for an example.

Reply:
I would suggest adding a comment to the code, particularly when the behavior of the HAPI and Blaze FHIR servers differs.